doc/releases/changelog-dev.md

```diff
@@ -474,6 +474,9 @@
 * Fixes a bug in `hamiltonian_expand` that produces incorrect output dimensions when shot vectors are combined with parameter broadcasting.
   [(#5494)](https://github.com/PennyLaneAI/pennylane/pull/5494)
 
+* Fixes a bug where `TorchLayer` does not work with shot vectors.
+  [(#5492)](https://github.com/PennyLaneAI/pennylane/pull/5492)
+
```

Comment:
Would you mind also adding the return type fix? :see_no_evil: Sorry for being picky about this, I think it's good to have this change tracked :)